packages/app/src/sessions.ts

```diff
@@ -236,6 +238,27 @@ class FatalError extends Error {}
 class TimeoutError extends Error {}
 const EventStreamContentType = 'text/event-stream';
 
+async function* streamToAsyncIterator<T>(
```

Comment:
very fancy. using iterator to abstract streaming while loop. it def makes code so much readable 💪